packages/components/src/components/tabs/tabs.jsx

```diff
@@ -4,168 +4,135 @@ import { withRouter } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import Tab from './tab.jsx';
 
-class Tabs extends React.Component {
-    constructor(props) {
-        super(props);
-
-        // if hash is in url, find which tab index correlates to it
-        const hash = this.props.location.hash.slice(1);
-        const hash_index = props.children.findIndex(child => child.props && child.props['data-hash'] === hash);
-        const has_hash = hash_index > -1;
-
-        // set active index to:
-        // 1. hash, if present
-        // 2. active_index prop, if present
-        // 3. default to first tab
-        const index_to_show = has_hash ? hash_index : props.active_index || 0;
-
-        // if no hash is in url but component has passed data-hash prop, set hash of the tab shown
-        if (!has_hash) {
-            const child_props = props.children[index_to_show].props;
-            const current_id = child_props && child_props['data-hash'];
-            if (current_id) {
-                this.updateUrl(current_id);
+const Tabs = ({
+    active_index,
+    bottom,
+    center,
+    children,
+    className,
+    fit_content,
+    header_fit_content,
+    history,
+    onTabItemClick,
+    should_update_hash,
+    single_tab_has_no_label,
+    top,
+}) => {
+    const [active_tab_index, setActiveTabIndex] = React.useState();
+    const [active_line_style, updateActiveLineStyle] = React.useState({});
+    const active_tab_ref = React.useRef();
+    const tabs_wrapper_ref = React.useRef();
+
+    React.useEffect(() => {
+        onClickTabItem(active_index || 0);
+    }, []);
+
+    React.useEffect(() => {
+        setActiveLineStyle();
+    }, [active_tab_index]);
```

Comment:
Could we add `setActiveLineStyle` to deps array?

Reply:
Salar, Thanks for the comment
I had checked this before but it not works fine in all cases and sometimes line_style doesn't work fine